pkg/common/file.go

```diff
@@ -40,23 +43,122 @@ func ReadConfig(filepath string) (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	config := Config{}
-	err = yaml.Unmarshal(data, &config)
+	yamlConfig := yaml.Node{}
+	err = yaml.Unmarshal(data, &yamlConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := getConfigOptions(yamlConfig)
 	if err != nil {
 		return nil, err
 	}
-	return &config, err
+
+	return config, err
 }
 
 // WriteConfig writes config to file
 // an error if the file was unable to be written to.
 func WriteConfig(c *Config, filepath string) error {
-	data, err := yaml.Marshal(c)
+	setConfigOptions(c)
+
+	data, err := yaml.Marshal(c.yamlData)
 	if err != nil {
 		return err
 	}
+
 	if err := os.MkdirAll(gofilepath.Dir(filepath), 0o755); err != nil {
 		return err
 	}
 	return ioutil.WriteFile(filepath, data, 0o644)
 }
+
+// Extracts config options from the yaml data which is loaded from file
+func getConfigOptions(yamlData yaml.Node) (*Config, error) {
+	config := Config{}
+	config.yamlData = &yamlData
+	contentLen := len(yamlData.Content[0].Content)
```

Comment:
What if `yamlData.Content == nil` or  `yamlData.Content[0].Content == nil`?

Reply:
This is currently a WIP. It is now unfortunately writing the tags of each value to file. I have opened issue https://github.com/go-yaml/yaml/issues/619 in go-yaml project.  Maybe someone might know also whats up?

Reply:
Resolved, incorrect tag on my part.